### PR TITLE
Fix usage of uninitialized Ui instance

### PR DESCRIPTION
**Pull request by gsoltis**
- removes `targetUI` with was initialized without anywhere to write
 - change usages of `targetUI` to `prefixedUI`, which is initialized
 - add an integration test for a script in a monorepo that returns an error (and now demonstrates integration tests for monorepos, not just single packages)

### CODE REVIEW

File: cli/integration_tests/monorepo_one_script_error/run.t [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Setup
+  $ . ${TESTDIR}/../setup.sh
+  $ . ${TESTDIR}/setup.sh $(pwd)
+
+Check error is properly report
```
Comment (mehulkar) [inline cli/integration_tests/monorepo_one_script_error/run.t:5]:
```suggestion
Check error is properly reported
```

File: cli/internal/run/run.go [hunk 2]
```diff
@@ -1122,7 +1120,7 @@ func (ec *execContext) exec(ctx gocontext.Context, packageTask *nodes.PackageTas
 	if err := closeOutputs(); err != nil {
 		ec.logError(progressLogger, "", err)
 	} else {
-		if err = taskCache.SaveOutputs(ctx, progressLogger, targetUI, int(duration.Milliseconds())); err != nil {
+		if err = taskCache.SaveOutputs(ctx, progressLogger, prefixedUI, int(duration.Milliseconds())); err != nil {
```
Comment (mehulkar) [inline cli/internal/run/run.go:1123]: for what it's worth, `progressLogger` and `targetUI` (now `prefixedUI`) are being used for pretty much the same UI. The only difference is the `progressLogger` messages aren't printed without `--verbosity=3` flag (depending on level). 

As far as I could tell `progressLogger` logs aren't written to disk or anything either? In which case, maybe it makes sense to just remove them all together?

**Review by mehulkar — APPROVED** [review]
Looks good

File: cli/internal/run/run.go [hunk 3]
```diff
@@ -997,9 +997,6 @@ func (ec *execContext) exec(ctx gocontext.Context, packageTask *nodes.PackageTas
 	// Setup tracer
 	tracer := ec.runState.Run(packageTask.TaskID)
 
-	// Create a logger
-	targetUI := &cli.BasicUi{}
-
```
Comment (nathanhammond) [inline cli/internal/run/run.go:1002]: This was an incomplete migration from here:
https://github.com/vercel/turborepo/pull/2126/files#diff-a79bb9f30997e3382bc1428116c8d0637778720674f22d669b2d7862607e535aR996

`prefixedUI` has the same construction as `targetUI` did originally and is vestigial at this point.

**Review by nathanhammond — APPROVED** [review]

File: cli/internal/run/run.go (hunk 3 above)
Comment (mehulkar) [reply cli/internal/run/run.go:1002]: it was intentional, i just didn't know what I was doing I guess 😢 . I thought `BasicUi` had what it needed to write logs by itself.

File: cli/internal/run/run.go (hunk 2 above)
Comment (nathanhammond) [reply cli/internal/run/run.go:1123]: I agree that `progressLogger` definitely needs to be cleaned up. It appears to be (mostly?) superfluous at this point. I don't have a good sense of how it came into existence though, so we should circle back to that.

**Pull request merged**